apps/veil/src/pages/portfolio/ui/shield-unshield.tsx

```diff
@@ -0,0 +1,228 @@
+import type { UnifiedAsset } from '@/pages/portfolio/api/use-unified-assets.ts';
+import { Button } from '@penumbra-zone/ui/Button';
+import { Tooltip } from '@penumbra-zone/ui/Tooltip';
+import { Widget } from '@skip-go/widget';
+import { useEffect, useState, useMemo } from 'react';
+import { X } from 'lucide-react';
+import { useRegistry } from '@/shared/api/registry.ts';
+import { getMetadata } from '@penumbra-zone/getters/value-view';
+
+/**
+ * Computes the IBC denom hash for Penumbra-1
+ * If the denom already starts with "ibc/", it's assumed to be already computed.
+ * If the denom starts with "transfer/", it's assumed to be a full trace path.
+ * Otherwise, constructs the trace path using channelId and denom.
+ */
+async function computeIbcDenom(denom: string, channelId: string): Promise<string> {
+  // Check if denom is already an IBC hash
+  if (denom.startsWith('ibc/')) {
+    return denom;
+  }
+  // Check if denom already is a full trace path
+  const ibcTraceStr = denom.startsWith('transfer/') ? denom : `transfer/${channelId}/${denom}`;
+
+  // Convert string to bytes using TextEncoder
+  const encoder = new TextEncoder();
+  const encodedString = encoder.encode(ibcTraceStr);
+
+  // Hash using SHA-256
+  const hashBuffer = await crypto.subtle.digest('SHA-256', encodedString);
+
+  // Convert to hex string and uppercase
+  const hashArray = Array.from(new Uint8Array(hashBuffer));
+  const hashHex = hashArray
+    .map(b => b.toString(16).padStart(2, '0'))
+    .join('')
+    .toUpperCase();
+
+  // Return in ibc/HASH format
+  return `ibc/${hashHex}`;
+}
+
+export function UnshieldButton({ asset }: { asset: UnifiedAsset }) {
+  const handleClick = () => {
+    console.warn('Unshield button clicked for asset:', asset.symbol);
+    // TODO: invoke dialog from minifront here or call unshielding function
+  };
+
+  return (
+    <Button onClick={handleClick} disabled={asset.shieldedBalances.length === 0}>
+      Unshield
+    </Button>
+  );
+}
+
+const theme = {
```

Comment:
I think it's better to link these colors by importing and using the vars of our theme.

Reply:
Done